lib/Email/MIME.pm

```diff
@@ -890,6 +895,21 @@ This method behaves like C<header_raw_pairs>, returning a list of field
 name/value pairs, but the values have been decoded to character strings, when
 possible.
 
+=method header_as_obj
+
+  my $first_obj = $email->header_as_obj($field);
+  my $nth_obj   = $email->header_as_obj($field, $index);
+  my @all_objs  = $email->header_as_obj($field);
+
+  my $nth_obj_of_class  = $email->header_as_obj($field, $index, $class);
+  my @all_objs_of_class = $email->header_as_obj($field, undef, $class);
+
+This method returns an object representation of the header value.  It instances
+new object via method C<from_mime_string> of specified class.  Input argument
+for that class method is list of the raw MIME-encoded values.  If class argument
+is not specified then class name is taken from the hash
+C<%Email::MIME::Header::header_to_class_map> via key field.
```

Comment:
Alternatives are:
* Introduce functions like: `register_class_map(header, class)` and `unregister_class_map(header, class)` (which will directly manipulate with that hash)
* Introduce function `header_to_class_map()` which will return reference to hash and user can directly modify it (or another class reimplement)
* Introduce our hash variable `%header_to_class_map` and user can directly use it

Basically all those alternatives provides needed API for own headers, just at different level of abstraction. I have no strong preference between for any of those. So if you think that second option (function which returns reference to that hash) is the best I will change my patches to it.